echidna/output/plot.py

```diff
@@ -23,31 +31,33 @@ def plot_projection(spectra, dimension, fig_num=1, show_plot=True):
 
     Args:
       spectra (:class:`echidna.core.spectra`): The spectra to plot.
-      dimension (int): The dimension to project the spectra onto.
+      dimension (string): The dimension to project the spectra onto.
+      fig_num (int, optional): The number of the figure. If you are
+        producing multiple spectral plots automatically, this can be
+        useful to ensure pyplot treats each plot as a separate figure.
+        Default is 1.
+      show_plot (bool, optional): Displays the plot if true. Default is True.
+
+    Returns:
+      matplotlib.pyplot.figure: Plot of the projection.
     """
-    fig = plt.figure(num=fig_num)
-    ax = fig.add_subplot(1, 1, 1)
-    if dimension == 0:
-        x = _produce_axis(spectra._energy_low, spectra._energy_high, spectra._energy_bins)
-        width = spectra._energy_width
-        plt.xlabel("Energy [MeV]")
-    elif dimension == 1:
-        x = _produce_axis(spectra._radial_low, spectra._radial_high, spectra._radial_bins)
-        width = spectra._radial_width
-        plt.xlabel("Radius [mm]")
-    elif dimension == 2:
-        x = _produce_axis(spectra._time_low, spectra._time_high, spectra._time_bins)
-        width = spectra._time_width
-        plt.xlabel("Time [yr]")
-    plt.ylabel("Count per %.2g bin" % width)
+    fig = plt.figure(fig_num)
+    axis = fig.add_subplot(1, 1, 1)
+    x = _produce_axis(spectra, dimension)
+    par = spectra.get_config().get_par(dimension)
+    width = par.get_width()
+    plt.xlabel("%s (%s)", (dimension, unit))
```

Comment:
A couple of errors during plotting. On this line `unit` is undefined, should be `par.get_unit()` I think